testing-internal/src/main/java/com/linecorp/armeria/internal/testing/AbstractMethodChainingTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.testing;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.reflections.ReflectionUtils;
+import org.reflections.Reflections;
+import org.reflections.adapters.JavaReflectionAdapter;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Makes sure most builder overrides all overridden methods with the correct return type.
+ */
+public abstract class AbstractMethodChainingTest {
+
+    @Test
+    void methodChaining() {
+        final String packageName = getClass().getPackage().getName();
+        findAllClasses(packageName).stream()
+                                   .map(ReflectionUtils::forName)
+                                   .filter(clazz -> clazz.getSimpleName().endsWith("Builder"))
+                                   .forEach(clazz -> {
+                                       final List<Method> methods = obtainMethods(clazz);
+                                       for (Method m : methods) {
+                                           try {
+                                               final Method overriddenMethod =
+                                                       clazz.getDeclaredMethod(m.getName(),
+                                                                               m.getParameterTypes());
+                                               assertThat(overriddenMethod.getReturnType()).isSameAs(clazz);
+                                           } catch (NoSuchMethodException e) {
+                                               // ignored
```

Comment:
Shouldn't we fail in this case? 🤔

Reply:
I think we can ignore this because we get the method through the parent class.